liquibase-bigquery/src/main/java/liquibase/snapshot/jvm/BigQueryForeignKeySnapshotGenerator.java

```diff
@@ -0,0 +1,128 @@
+            ForeignKey foreignKey = null;
+            for (Map<String, ?> resultMap : results) {
+                String foreignKeyName = Objects.toString(resultMap.get("CONSTRAINT_NAME"));
+
+                Table foreignKeyTable = new Table().setName(Objects.toString(resultMap.get("FOREIGN_KEY_TABLE")));
+                foreignKeyTable.setSchema(new Schema(new Catalog(Objects.toString(resultMap.get("FOREIGN_KEY_TABLE_CATALOG"))),
+                        Objects.toString(resultMap.get("FOREIGN_KEY_TABLE_SCHEMA"))));
+
+                Table primaryKeyTable = new Table().setName(Objects.toString(resultMap.get("PRIMARY_KEY_TABLE")));
+                primaryKeyTable.setSchema(
+                        new Schema(new Catalog(Objects.toString(resultMap.get("PRIMARY_KEY_TABLE_CATALOG"))),
+                                Objects.toString(resultMap.get("PRIMARY_KEY_TABLE_SCHEMA"))));
+                Column fkColumn = new Column(Objects.toString(resultMap.get("FOREIGN_KEY_COLUMN"))).setRelation(foreignKeyTable);
+                Column pkColumn = new Column(Objects.toString(resultMap.get("PRIMARY_KEY_COLUMN"))).setRelation(primaryKeyTable);
+
+                if (foreignKey != null) {
+                    if (!foreignKey.getForeignKeyColumns().contains(fkColumn)) {
+                        foreignKey.addForeignKeyColumn(fkColumn);
+                    }
+                    if (!foreignKey.getPrimaryKeyColumns().contains(pkColumn)) {
+                        foreignKey.addPrimaryKeyColumn(pkColumn);
+                    }
+                } else {
+                    foreignKey = new ForeignKey(foreignKeyName);
+
+                    foreignKey.setForeignKeyTable(foreignKeyTable);
+                    foreignKey.setPrimaryKeyTable(primaryKeyTable);
+
+                    foreignKey.addForeignKeyColumn(fkColumn);
+                    foreignKey.addPrimaryKeyColumn(pkColumn);
+
+                }
+            }
+            return foreignKey;
+        }
+        return null;
+    }
+
+    @Override
+    protected void addTo(DatabaseObject foundObject, DatabaseSnapshot snapshot) throws DatabaseException {
+        if (!snapshot.getSnapshotControl().shouldInclude(ForeignKey.class)) {
+            return;
+        }
+        if (foundObject instanceof Table) {
+            Table table = (Table) foundObject;
+            Database database = snapshot.getDatabase();
+            Schema schema = table.getSchema();
+
+            CatalogAndSchema catalogAndSchema = (new CatalogAndSchema(schema.getCatalogName(), schema.getName())).customize(database);
+            String jdbcSchemaName = database.correctObjectName(((AbstractJdbcDatabase) database).getJdbcSchemaName(catalogAndSchema), Schema.class);
+            String query = String.format("SELECT CONSTRAINT_NAME FROM %s.%s.TABLE_CONSTRAINTS WHERE table_name='%s' AND table_schema='%s' AND " +
+                            "table_catalog='%s' AND CONSTRAINT_TYPE='FOREIGN KEY';",
```

Comment:
Another PreparedStatement for table.getName(), schema.getName(), schema.getCatalogName()